File: cppfastlogging/examples/get_server_addresses_ports.cpp
#include "h/cppfastlogging.hpp"
#include <cstdio>

using namespace logging;

int main(void)
{
    Logging logging_server(DEBUG, "LOGSRV");
    logging_server.add_writer_config(ConsoleWriterConfig(DEBUG, true));
    ServerConfig srv(DEBUG, "127.0.0.1");
    logging_server.add_writer_config(srv);
    logging_server.set_root_writer_config(srv);
    logging_server.sync_all(5.0);

    const Cu32u16Vec_t *ports = logging_server.get_server_ports();
    if (ports) {
        printf("ports->cnt=%d\n", ports->cnt);
        for (uint32_t i = 0; i < ports->cnt; i++) {
            printf("ports->key[%u]=%u\n",   i, ports->keys[i]);
            printf("ports->value[%u]=%u\n", i, ports->values[i]);
        }
    }
    const Cu32StringVec_t *addresses = logging_server.get_server_addresses();
    if (addresses) {
        printf("addresses->cnt=%d\n", addresses->cnt);
        for (uint32_t i = 0; i < addresses->cnt; i++) {
            printf("addresses->key[%u]=%u\n",   i, addresses->keys[i]);
            printf("addresses->value[%u]=%s\n", i, addresses->values[i]);
        }
    }
    const Cu32StringVec_t *ap = logging_server.get_server_addresses_ports();
    if (ap) {
        printf("addresses_ports->cnt=%d\n", ap->cnt);
        for (uint32_t i = 0; i < ap->cnt; i++) {
            printf("addresses_ports->key[%u]=%u\n",   i, ap->keys[i]);
            printf("addresses_ports->value[%u]=%s\n", i, ap->values[i]);
        }
    }
    logging_server.info("Info Message");
    logging_server.sync_all(1.0);
    printf("-------- Finished --------\n");
    return 0;
}

File: fastlogging/examples/get_server_addresses_ports.rs
use std::{thread, time::Duration};

use fastlogging::{
    ConsoleWriterConfig, DEBUG, EncryptionMethod, Logging, LoggingError, ServerConfig,
};

fn main() -> Result<(), LoggingError> {
    // Server
    let mut logging_server = Logging::new_unboxed(
        DEBUG,
        "LOGSRV",
        Some(vec![
            ConsoleWriterConfig::new(DEBUG, true).into(),
            ServerConfig::new(DEBUG, "127.0.0.1", EncryptionMethod::NONE).into(),
        ]),
        None,
        None,
    )?;
    // Set root writer
    logging_server.set_root_writer_config(
        &ServerConfig::new(DEBUG, "127.0.0.1", EncryptionMethod::NONE).into(),
    )?;
    //logging_server.set_debug(3);
    logging_server.sync_all(5.0).unwrap();
    // Show addresses and ports
    let ports = logging_server.get_server_ports();
    println!("ports={ports:#?}");
    let addresses = logging_server.get_server_addresses();
    println!("addresses={addresses:#?}");
    let address_ports = logging_server.get_server_addresses_ports();
    println!("address_ports={address_ports:#?}");
    // Test logging
    println!("Send logs");
    logging_server.trace("Trace Message".to_string()).unwrap();
    logging_server.debug("Debug Message".to_string()).unwrap();
    logging_server.info("Info Message".to_string()).unwrap();
    logging_server
        .success("Success Message".to_string())
        .unwrap();
    logging_server
        .warning("Warning Message".to_string())
        .unwrap();
    logging_server.error("Error Message".to_string()).unwrap();
    logging_server.fatal("Fatal Message".to_string()).unwrap();

    logging_server.sync_all(1.0)?;
    // Give client some time to send the log messages
    thread::sleep(Duration::from_millis(50));
    println!("Shutdown Loggers");
    logging_server.shutdown(false).unwrap();
    println!("-------- Finished --------");
    Ok(())
}
